File: t/PACSFramework/CS/StorageServerManager/UI/ServerMainMenu/ServerMainMenu.cs
// *************************************************************
// Copyright (c) 1991-2019 LEAD Technologies, Inc.              
// All Rights Reserved.                                         
// *************************************************************
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Diagnostics;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using Leadtools.Demos.StorageServer.UI.AdministrativeSettings.Users;

namespace Leadtools.Demos.StorageServer.UI
{
   public partial class ServerMainMenu : MenuStrip
   {
      #region Public
         
         #region Methods
         
            public ServerMainMenu ( )
            {
               InitializeComponent ( ) ;

               Init ( ) ;

               RegisterEvents ( ) ;
            }
         
         #endregion
         
         #region Properties
            
         #endregion
         
         #region Data Types Definition
            
         #endregion
         
         #region Callbacks
         
            public event EventHandler ExitApplication ;
            public event EventHandler ShowSettings ;
            public event EventHandler ShowAbout ;
            
         #endregion
         
      #endregion
      
      #region Protected
         
         #region Methods
            
         #endregion
         
         #region Properties
            
         #endregion
         
         #region Data Types Definition
            
         #endregion
         
      #endregion
      
      #region Private
         
         #region Methods

            private void Init ( )
            {
               _file          = new ToolStripMenuItem ( "&File" ) ;
               _tools         = new ToolStripMenuItem ( "&Tools" ) ;
               _help          = new ToolStripMenuItem ( "&Help" ) ;
               _fileExit      = new ToolStripMenuItem ( "E&xit" ) ;
               _toolsSettings = new ToolStripMenuItem (  "&Settings..." ) ;
               _helpAbout     = new ToolStripMenuItem ( "&About" ) ;


               _file.DropDownItems.Add  ( _fileExit ) ;
               _tools.DropDownItems.Add ( _toolsSettings ) ;
               _help.DropDownItems.Add  ( _helpAbout ) ;

               this.Items.Add (_file ) ;
               this.Items.Add ( _tools ) ;
               this.Items.Add ( _help ) ;

               _toolsSettings.Enabled = UserManager.User.IsAdmin() || UserManager.User.IsAuthorized ( UserPermissions.CanChangeServerSettings ) ;
            }

            private void RegisterEvents ( )
            {
               _fileExit.Click      += new EventHandler ( _fileExit_Click ) ;
               _toolsSettings.Click += new EventHandler ( _toolsSettings_Click ) ;
               _helpAbout.Click     += new EventHandler ( _helpAbout_Click ) ;
            }
            
         #endregion
         
         #region Properties
            
         #endregion
         
         #region Events
               
            void _toolsSettings_Click ( object sender, EventArgs e )
            {
               try
               {
                  if ( null != ShowSettings )
                  {
                     ShowSettings ( this, e ) ;
                  }
               }
               catch ( Exception ex ) 
               {
                  Messager.ShowError ( this, ex ) ;
               }
            }

            void _fileExit_Click ( object sender, EventArgs e )
            {
               try
               {
                  if ( null != ExitApplication ) 
                  {
                     ExitApplication ( this, e ) ;
                  }
               }
               catch ( Exception ex ) 
               {
                  Messager.ShowError ( this, ex ) ;
               }
            }

            void _helpAbout_Click(object sender, EventArgs e)
            {
               try
               {
                  if ( null != ShowAbout ) 
                  {
                     ShowAbout ( this, e ) ;
                  }
               }
               catch ( Exception ex ) 
               {
                  Messager.ShowError ( this, ex ) ;
               }
            }
         
         #endregion
         
         #region Data Members
         
            private ToolStripMenuItem _file ; 
            private ToolStripMenuItem _tools ; 
            private ToolStripMenuItem _help ; 
            private ToolStripMenuItem _fileExit ;
            private ToolStripMenuItem _toolsSettings ; 
            private ToolStripMenuItem _helpAbout ; 
            
         #endregion
         
         #region Data Types Definition
            
         #endregion
         
      #endregion
      
      #region internal
         
         #region Methods
            
         #endregion
         
         #region Properties
            
         #endregion
         
         #region Data Types Definition
            
         #endregion
         
         #region Callbacks
            
         #endregion
         
      #endregion
   }
   
}

File: t/PACSFramework/VB/StorageServerManager/UI/ServerMainMenu/ServerMainMenu.vb
' *************************************************************
' Copyright (c) 1991-2019 LEAD Technologies, Inc.              
' All Rights Reserved.                                         
' *************************************************************
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Diagnostics
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports Leadtools.Demos.StorageServer.UI.AdministrativeSettings.Users

Namespace Leadtools.Demos.StorageServer.UI
   Partial Public Class ServerMainMenu : Inherits MenuStrip
#Region "Public"

#Region "Methods"

      Public Sub New()
         InitializeComponent()

         Init()

         RegisterEvents()
      End Sub

#End Region

#Region "Properties"

#End Region

#Region "Data Types Definition"

#End Region

#Region "Callbacks"

      Public Event ExitApplication As EventHandler
      Public Event ShowSettings As EventHandler
      Public Event ShowAbout As EventHandler

#End Region

#End Region

#Region "Protected"

#Region "Methods"

#End Region

#Region "Properties"

#End Region

#Region "Data Types Definition"

#End Region

#End Region

#Region "Private"

#Region "Methods"

      Private Sub Init()
         _file = New ToolStripMenuItem("&File")
         _tools = New ToolStripMenuItem("&Tools")
         _help = New ToolStripMenuItem("&Help")
         _fileExit = New ToolStripMenuItem("E&xit")
         _toolsSettings = New ToolStripMenuItem("&Settings...")
         _helpAbout = New ToolStripMenuItem("&About")


         _file.DropDownItems.Add(_fileExit)
         _tools.DropDownItems.Add(_toolsSettings)
         _help.DropDownItems.Add(_helpAbout)

         Me.Items.Add(_file)
         Me.Items.Add(_tools)
         Me.Items.Add(_help)

         _toolsSettings.Enabled = UserManager.User.IsAdmin() OrElse UserManager.User.IsAuthorized(UserPermissions.CanChangeServerSettings)
      End Sub

      Private Sub RegisterEvents()
         AddHandler _fileExit.Click, AddressOf _fileExit_Click
         AddHandler _toolsSettings.Click, AddressOf _toolsSettings_Click
         AddHandler _helpAbout.Click, AddressOf _helpAbout_Click
      End Sub

#End Region

#Region "Properties"

#End Region

#Region "Events"

      Private Sub _toolsSettings_Click(ByVal sender As Object, ByVal e As EventArgs)
         Try
            If Not Nothing Is ShowSettingsEvent Then
               RaiseEvent ShowSettings(Me, e)
            End If
         Catch ex As Exception
            Messager.ShowError(Me, ex)
         End Try
      End Sub

      Private Sub _fileExit_Click(ByVal sender As Object, ByVal e As EventArgs)
         Try
            If Not Nothing Is ExitApplicationEvent Then
               RaiseEvent ExitApplication(Me, e)
            End If
         Catch ex As Exception
            Messager.ShowError(Me, ex)
         End Try
      End Sub

      Private Sub _helpAbout_Click(ByVal sender As Object, ByVal e As EventArgs)
         Try
            If Not Nothing Is ShowAboutEvent Then
               RaiseEvent ShowAbout(Me, e)
            End If
         Catch ex As Exception
            Messager.ShowError(Me, ex)
         End Try
      End Sub

#End Region

#Region "Data Members"

      Private _file As ToolStripMenuItem
      Private _tools As ToolStripMenuItem
      Private _help As ToolStripMenuItem
      Private _fileExit As ToolStripMenuItem
      Private _toolsSettings As ToolStripMenuItem
      Private _helpAbout As ToolStripMenuItem

#End Region

#Region "Data Types Definition"

#End Region

#End Region

#Region "internal"

#Region "Methods"

#End Region

#Region "Properties"

#End Region

#Region "Data Types Definition"

#End Region

#Region "Callbacks"

#End Region

#End Region
   End Class

End Namespace
